auto_dev/services/eject/index.py

```diff
@@ -0,0 +1,232 @@
+        packages_base = Path("..") / "packages"
+        possible_paths = [
+            packages_base / self.config.fork_id.author / "agents" / self.config.fork_id.name,
+            packages_base / "agents" / self.config.fork_id.author / self.config.fork_id.name,
+            packages_base / self.config.component_type / self.config.fork_id.author / self.config.fork_id.name,
+            packages_base / self.config.fork_id.author / "agents" / Path.cwd().name,
+        ]
+
+        for path in possible_paths:
+            if path.exists():
+                self.logger.info(f"Cleaning up existing directory: {path}")
+                shutil.rmtree(path)
+
+        packages_base.mkdir(parents=True, exist_ok=True)
+
+    def handle_dependencies(self, deps: List[PublicId]) -> List[PublicId]:
+        """Handle component dependencies."""
+        if self.config.skip_dependencies:
+            return []
+
+        ejected_deps = []
+        for dep in deps:
+            ejected_dep = self._try_eject_dependency(dep)
+            if ejected_dep:
+                ejected_deps.append(ejected_dep)
+        return ejected_deps
+
+    def _try_eject_dependency(self, dep: PublicId) -> Optional[PublicId]:
+        """Try to eject a single dependency."""
+        for component_type in ComponentType:
+            component_path = self.config.base_path / f"{component_type.value}s" / dep.author / dep.name
+            if not component_path.exists():
+                continue
+
+            if self.eject_component(dep, self.config.fork_id.author, dep.name, component_type.value):
+                self.logger.info(f"Successfully ejected dependency {dep}")
+                return dep
+        return None
+
+    def publish_and_lock(self) -> bool:
+        """Publish packages and run lock command."""
+        # need to implement publish service function and use here
+        success, _ = self.run_command("aea publish --push-missing --local")
+        if not success:
+            raise ValueError("Failed to publish packages")
+
+        with change_dir(".."):
+            _, exit_code = self.run_command("yes dev | autonomy packages lock", shell=True)
+            if exit_code not in [0, 1]:
+                raise ValueError(f"Packages lock failed with exit code {exit_code}")
+        return True
+
+    def eject(self) -> List[PublicId]:
+        """
+        Eject a component and its dependents.
+
+        Returns:
+            List of ejected component IDs
+
+        Raises:
+            ValueError: If dependencies are skipped but not already ejected
+        """
+        try:
+            ejected_deps = self._handle_component_dependencies()
+
+            # Eject main component
+            if not self.eject_component(
+                self.config.public_id, self.config.fork_id.author, self.config.fork_id.name, self.config.component_type
+            ):
+                return []
+
+            self._update_and_cleanup()
+            self.publish_and_lock()
```

Comment:
this should be in the publish class

Reply:
yeah i think I said this earlier but there is a third ticker after this to combine this and the publish logic together. don't want to combine PRs